src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -3354,7 +3354,8 @@ public static int SplitAny(this ReadOnlySpan<char> source, Span<Range> destinati
 
             // If the separators list is empty, whitespace is used as separators.  In that case, we want to ignore TrimEntries if specified,
             // since TrimEntries also impacts whitespace.
-            if (separators.IsEmpty)
+            // The TrimEntries flag must be left intact if we are constrained by count because we need to process last substring.
```

Comment:
Nit: here and below, please remove the newline between the existing sentence and the new one

Reply:
Will do.